### ISSUE TITLE
Failed to get secret "istio-ca-secret" thus istiod pod's readiness probe fails on EKS

### ISSUE THREAD

**Issue by hasakura12** [istio/istio#24009]
**Bug description**
Failed to get secret "istio-ca-secret" thus istiod pod's readiness probe fails.

**Expected behavior**
istiod pod should be running on EKS

**Steps to reproduce the bug**
Install `default` profile (tried `demo` as well, no luck)
```
istioctl manifest apply --set profile=default
```

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

**How was Istio installed?**
```
curl -L https://istio.io/downloadIstio | sh -
cd istio-1.5.4
echo "export PATH=$PWD/bin:$PATH" >> ~/.bash_profile
```

Made sure no network policy
```
$ kubectl get networkpolicy --all-namespaces
No resources found.
```

Namespace has a label
```
kubectl describe namespace staging
Name:         staging
Labels:       istio-injection=enabled
Annotations:  <none>
```

Install `default` profile (tried `demo` as well, no luck)
```
istioctl manifest apply --set profile=default
```

Output
```
- Applying manifest for component Base...
✔ Finished applying manifest for component Base.
- Applying manifest for component Pilot...
✔ Finished applying manifest for component Pilot.
  Waiting for resources to become ready...
  Waiting for resources to become ready...
  Waiting for resources to become ready...
  Waiting for resources to become ready...
  Waiting for resources to become ready...
  Waiting for resources to become ready...
  Waiting for resources to become ready...
  Waiting for resources to become ready...
  Waiting for resources to become ready...
```

istiod pod details
```
$ kubectl describe pod/istiod-74c6d9ccdb-xhrdd -n istio-system

Events:
  Type     Reason     Age                From                                  Message
  ----     ------     ----               ----                                  -------
  Normal   Scheduled  79s                default-scheduler                     Successfully assigned istio-system/istiod-74c6d9ccdb-xhrdd to ip-XXX.ec2.internal
  Normal   Pulled     17s (x2 over 78s)  kubelet, ip-XXX.ec2.internal  Container image "docker.io/istio/pilot:1.5.4" already present on machine
  Normal   Created    17s (x2 over 78s)  kubelet, ip-XXX.ec2.internal  Created container discovery
  Normal   Started    17s (x2 over 77s)  kubelet, ip-XXX.ec2.internal  Started container discovery
  Warning  Unhealthy  5s (x12 over 65s)  kubelet, ip-XXX.ec2.internal  Readiness probe failed: Get http://10.1.103.202:8080/ready: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

istiod pod's log
```
$ kubectl logs pod/istiod-74c6d9ccdb-xhrdd  -n istio-system

Flag --disable-install-crds has been deprecated, Setting this flag has no effect. Install CRD definitions directly or with the operator
2020-05-21T10:42:24.961753Z     info    FLAG: --appNamespace=""
2020-05-21T10:42:24.961788Z     info    FLAG: --clusterRegistriesNamespace=""
2020-05-21T10:42:24.961800Z     info    FLAG: --configDir=""
2020-05-21T10:42:24.961810Z     info    FLAG: --consulserverURL=""
2020-05-21T10:42:24.961820Z     info    FLAG: --ctrlz_address="localhost"
2020-05-21T10:42:24.961833Z     info    FLAG: --ctrlz_port="9876"
2020-05-21T10:42:24.961846Z     info    FLAG: --disable-install-crds="true"
2020-05-21T10:42:24.961857Z     info    FLAG: --domain="cluster.local"
2020-05-21T10:42:24.961867Z     info    FLAG: --grpcAddr=":15010"
2020-05-21T10:42:24.961877Z     info    FLAG: --help="false"
2020-05-21T10:42:24.961886Z     info    FLAG: --httpAddr=":8080"
2020-05-21T10:42:24.961896Z     info    FLAG: --httpsAddr=":15017"
2020-05-21T10:42:24.961908Z     info    FLAG: --keepaliveInterval="30s"
2020-05-21T10:42:24.961918Z     info    FLAG: --keepaliveMaxServerConnectionAge="30m0s"
2020-05-21T10:42:24.961927Z     info    FLAG: --keepaliveTimeout="10s"
2020-05-21T10:42:24.961936Z     info    FLAG: --kubeconfig=""
2020-05-21T10:42:24.961945Z     info    FLAG: --log_as_json="false"
2020-05-21T10:42:24.961954Z     info    FLAG: --log_caller=""
2020-05-21T10:42:24.961964Z     info    FLAG: --log_output_level="default:info"
2020-05-21T10:42:24.961974Z     info    FLAG: --log_rotate=""
2020-05-21T10:42:24.961987Z     info    FLAG: --log_rotate_max_age="30"
2020-05-21T10:42:24.962042Z     info    FLAG: --log_rotate_max_backups="1000"
2020-05-21T10:42:24.962054Z     info    FLAG: --log_rotate_max_size="104857600"
2020-05-21T10:42:24.962066Z     info    FLAG: --log_stacktrace_level="default:none"
2020-05-21T10:42:24.962091Z     info    FLAG: --log_target="[stdout]"
2020-05-21T10:42:24.962107Z     info    FLAG: --mcpInitialConnWindowSize="1048576"
2020-05-21T10:42:24.962117Z     info    FLAG: --mcpInitialWindowSize="1048576"
2020-05-21T10:42:24.962127Z     info    FLAG: --mcpMaxMsgSize="4194304"
2020-05-21T10:42:24.962138Z     info    FLAG: --meshConfig="/etc/istio/config/mesh"
2020-05-21T10:42:24.962152Z     info    FLAG: --monitoringAddr=":15014"
2020-05-21T10:42:24.962163Z     info    FLAG: --namespace=""
2020-05-21T10:42:24.962176Z     info    FLAG: --networksConfig="/etc/istio/config/meshNetworks"
2020-05-21T10:42:24.962193Z     info    FLAG: --plugins="[authn,authz,health,mixer]"
2020-05-21T10:42:24.962206Z     info    FLAG: --profile="true"
2020-05-21T10:42:24.962223Z     info    FLAG: --registries="[Kubernetes]"
2020-05-21T10:42:24.962239Z     info    FLAG: --resync="1m0s"
2020-05-21T10:42:24.962250Z     info    FLAG: --secureGrpcAddr=":15011"
2020-05-21T10:42:24.962260Z     info    FLAG: --trust-domain="cluster.local"
2020-05-21T10:42:24.962321Z     info    Primary Cluster name: Kubernetes
2020-05-21T10:42:24.965976Z     info    mesh configuration: {
    "disablePolicyChecks": true,
    "disableMixerHttpReports": true,
    "proxyListenPort": 15001,
    "connectTimeout": "1s",
    "protocolDetectionTimeout": "0.100s",
    "ingressClass": "istio",
    "ingressService": "istio-ingressgateway",
    "ingressControllerMode": "STRICT",
    "enableTracing": true,
    "defaultConfig": {
        "configPath": "/etc/istio/proxy",
        "binaryPath": "/usr/local/bin/envoy",
        "serviceCluster": "istio-proxy",
        "drainDuration": "45s",
        "parentShutdownDuration": "60s",
        "discoveryAddress": "istiod.istio-system.svc:15012",
        "connectTimeout": "10s",
        "proxyAdminPort": 15000,
        "statNameLength": 189,
        "concurrency": 2,
        "tracing": {
            "zipkin": {
                "address": "zipkin.istio-system:9411"
            }
        },
        "envoyAccessLogService": {

        },
        "envoyMetricsService": {

        }
    },
    "outboundTrafficPolicy": {
        "mode": "ALLOW_ANY"
    },
    "sdsUdsPath": "unix:/etc/istio/proxy/SDS",
    "enableAutoMtls": true,
    "trustDomain": "cluster.local",
    "trustDomainAliases": [
    ],
    "defaultServiceExportTo": [
        "*"
    ],
    "defaultVirtualServiceExportTo": [
        "*"
    ],
    "defaultDestinationRuleExportTo": [
        "*"
    ],
    "rootNamespace": "istio-system",
    "localityLbSetting": {
        "enabled": true
    },
    "dnsRefreshRate": "5s",
    "reportBatchMaxEntries": 100,
    "reportBatchMaxTime": "1s",
    "certificates": [
    ]
}
2020-05-21T10:42:24.966025Z     info    version: 1.5.4-f27639c7369f8a6ef144ed8768f8acb27566d999-Clean
2020-05-21T10:42:24.966916Z     info    flags: {
   "DiscoveryOptions": {
      "HTTPAddr": ":8080",
      "HTTPSAddr": ":15017",
      "GrpcAddr": ":15010",
      "SecureGrpcAddr": ":15011",
      "SecureGrpcDNSAddr": "",
      "MonitoringAddr": ":15014",
      "EnableProfiling": true
   },
   "InjectionOptions": {
      "InjectionDirectory": "./var/lib/istio/inject"
   },
   "PodName": "istiod-74c6d9ccdb-xhrdd",
   "Namespace": "istio-system",
   "Revision": "",
   "ServiceAccountName": "istio-pilot-service-account",
   "Mesh": {
      "ConfigFile": "/etc/istio/config/mesh",
      "MixerAddress": ""
   },
   "Config": {
      "ControllerOptions": {
         "WatchedNamespace": "",
         "ResyncPeriod": 60000000000,
         "DomainSuffix": "cluster.local",
         "ClusterID": "",
         "Metrics": null,
         "XDSUpdater": null,
         "TrustDomain": "cluster.local",
         "NetworksWatcher": null,
         "EndpointMode": 0
      },
      "ClusterRegistriesNamespace": "istio-system",
      "KubeConfig": "",
      "FileDir": "",
      "DistributionCacheRetention": 60000000000,
      "DisableInstallCRDs": true,
      "DistributionTrackingEnabled": true
   },
   "Service": {
      "Registries": [
         "Kubernetes"
      ],
      "Consul": {
         "ServerURL": ""
      }
   },
   "MeshConfig": null,
   "NetworksConfigFile": "/etc/istio/config/meshNetworks",
   "CtrlZOptions": {
      "Port": 9876,
      "Address": "localhost"
   },
   "Plugins": [
      "authn",
      "authz",
      "health",
      "mixer"
   ],
   "MCPMaxMessageSize": 4194304,
   "MCPInitialWindowSize": 1048576,
   "MCPInitialConnWindowSize": 1048576,
   "KeepaliveOptions": {
      "Time": 30000000000,
      "Timeout": 10000000000,
      "MaxServerConnectionAge": 1800000000000,
      "MaxServerConnectionAgeGrace": 10000000000
   },
   "ForceStop": false,
   "BasePort": 15000
}
2020-05-21T10:42:24.967152Z     info    mesh networks configuration: {
   "networks": {
   }
}
2020-05-21T10:42:24.967215Z     info    No certificates specified, skipping DNS certificate controller
2020-05-21T10:42:24.968516Z     info    CRD controller watching namespaces ""
2020-05-21T10:42:24.968873Z     info    Ingress controller watching namespaces ""
2020-05-21T10:42:24.969046Z     warn    Config Store &{0xc0007ae240 cluster.local 0xc0000d0dc0 0xc00061b740 0xc00067afc0 []} cannot track distribution in aggregate
2020-05-21T10:42:24.969081Z     info    Adding Kubernetes registry adapter
2020-05-21T10:42:24.969098Z     info    Service controller watching namespace "" for services, endpoints, nodes and pods, refresh 1m0s
2020-05-21T10:42:24.969260Z     info    JWT policy is third-party-jwt
2020-05-21T10:42:24.969345Z     info    Use self-signed certificate as the CA certificate
2020-05-21T10:42:54.970100Z     info    pkica   Failed to get secret (error: Get https://172.20.0.1:443/api/v1/namespaces/istio-system/secrets/istio-ca-secret: dial tcp 172.20.0.1:443: i/o timeout), will create one
2020-05-21T10:43:25.096662Z     error   pkica   Failed to write secret to CA (error: Post https://172.20.0.1:443/api/v1/namespaces/istio-system/secrets: dial tcp 172.20.0.1:443: i/o timeout). Abort.
2020-05-21T10:43:25.096772Z     error   failed to create discovery service: enableCA: failed to create a self-signed Citadel: failed to create CA due to secret write error
Error: failed to create discovery service: enableCA: failed to create a self-signed Citadel: failed to create CA due to secret write error
```

This line seems alerming to me `info    pkica   Failed to get secret (error: Get https://172.20.0.1:443/api/v1/namespaces/istio-system/secrets/istio-ca-secret: dial tcp 172.20.0.1:443: i/o timeout), will create one`. But with `default` profile, citadel is `false`, so not sure why it's even trying to create root CA.

Default profile's config
```
$ istioctl profile dump default

apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  addonComponents:
    grafana:
      enabled: false
      k8s:
        replicaCount: 1
    istiocoredns:
      enabled: false
    kiali:
      enabled: false
      k8s:
        replicaCount: 1
    prometheus:
      enabled: true
      k8s:
        replicaCount: 1
    tracing:
      enabled: false
  components:
    base:
      enabled: true
    citadel:
      enabled: false
      k8s:
        strategy:
          rollingUpdate:
            maxSurge: 100%
            maxUnavailable: 25%
    cni:
      enabled: false
```

Configmap
```
$ kubectl get configmap --all-namespaces | grep istio

istio-system   istio                                3      3m9s
istio-system   istio-sidecar-injector               2      3m5s
istio-system   pilot-envoy-config                   1      3m3s
```

Secret
```
$ kubectl get secret --all-namespaces | grep istio
istio-system      default-token-7cr7s                                     kubernetes.io/service-account-token   3      4m49s
istio-system      istio-pilot-service-account-token-c5b5n                 kubernetes.io/service-account-token   3      3m39s
istio-system      istio-reader-service-account-token-vbfbj                kubernetes.io/service-account-token   3      3m59s
```

**Environment where bug was observed (cloud vendor, OS, etc)**
```
istioctl version
2020-05-21T10:37:28.719376Z     warn    will use `--remote=false` to retrieve version info due to `no Istio pods in namespace "istio-system"`
1.5.4

kubectl version
Client Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.10-eks-bac369", GitCommit:"bac3690554985327ae4d13e42169e8b1c2f37226", GitTreeState:"clean", BuildDate:"2020-02-21T23:37:18Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.8-eks-e16311", GitCommit:"e163110a04dcb2f39c3325af96d019b4925419eb", GitTreeState:"clean", BuildDate:"2020-03-27T22:37:12Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by howardjohn** [comment]
Citadel is disabled but Istiod acts as the CA, so that is expected. So the issue here is that we cannot connect to the API server it seems

**Comment by hasakura12** [comment]
So does that mean by default SSL is enabled? I thought it's disabled unless citadel is enabled.

**Comment by lei-tang** [comment]
When you run "istioctl manifest apply --set profile=default" on other platforms, does this problem still occur?

**Comment by rajivml** [comment]
@howardjohn  hi, so what's the solution to fix this, we are trying to setup istio-1.5.6 on AKS and we are facing this exact same issue and am blocked on this 

```
mkdir ~/istio
cd ~/istio
ISTIO_VERSION=1.5.6
curl -sL "https://github.com/istio/istio/releases/download/$ISTIO_VERSION/istio-$ISTIO_VERSION-linux.tar.gz" | tar xz
cd istio-$ISTIO_VERSION
sudo cp ./bin/istioctl /usr/local/bin/istioctl
sudo chmod +x /usr/local/bin/istioctl

istioctl manifest apply --logtostderr \
    --set installPackagePath=./install/kubernetes/operator \
    --set values.gateways.istio-ingressgateway.sds.enabled=true \
    --set values.global.proxy.resources.limits.cpu=100m \
    --set values.global.proxy.resources.limits.memory=128Mi \
    --set values.sidecarInjectorWebhook.rewriteAppHTTPProbe=true \
    --set values.global.disablePolicyChecks=false \
    --set values.global.configValidation=true \
    --set values.prometheus.retention=1d \
    --set values.global.k8sIngress.enabled=true \
    --set values.global.k8sIngress.enableHttps=true \
    --set addonComponents.grafana.enabled=true \
    --set meshConfig.accessLogFile="/dev/stdout" \
    --set values.gateways.istio-ingressgateway.externalTrafficPolicy=Local

```
```
rajivchodisetti@DESKTOP-LOUPTI1:~/istio/istio-1.5.6$ kubectl -n istio-system get pods
NAME                                    READY   STATUS              RESTARTS   AGE
grafana-5cc7f86765-dhzd2                1/1     Running             0          32m
istio-ingressgateway-7d99cd78c4-79kc4   0/1     ContainerCreating   0          133m
istio-ingressgateway-c58879b5c-j4l4j    0/1     ContainerCreating   0          32m
istiod-7dbc6b66fc-79m6x                 0/1     CrashLoopBackOff    27         143m
istiod-87ddb64b8-65kmn                  0/1     Running             11         42m
prometheus-68ddb58568-64rf9             0/2     ContainerCreating   0          32m
prometheus-7746f68f8c-8jzw5             0/2     ContainerCreating   0          133m
```

```
kubectl -n istio-system logs -f istiod-87ddb64b8-65kmn

2020-07-28T17:04:58.333189Z     info    No certificates specified, skipping DNS certificate controller
2020-07-28T17:04:58.334079Z     info    CRD controller watching namespaces ""
2020-07-28T17:04:58.334292Z     info    Ingress controller watching namespaces ""
2020-07-28T17:04:58.334398Z     warn    Config Store &{0xc000259b00 cluster.local 0xc000256840 0xc000868960 0xc00085cf30 []} cannot track distribution in aggregate
2020-07-28T17:04:58.334416Z     info    Adding Kubernetes registry adapter
2020-07-28T17:04:58.334423Z     info    Service controller watching namespace "" for services, endpoints, nodes and pods, refresh 1m0s
2020-07-28T17:04:58.334482Z     info    JWT policy is third-party-jwt
2020-07-28T17:04:58.334541Z     info    Use self-signed certificate as the CA certificate
2020-07-28T17:05:28.335103Z     info    pkica   Failed to get secret (error: Get https://10.0.0.1:443/api/v1/namespaces/istio-system/secrets/istio-ca-secret: dial tcp 10.0.0.1:443: i/o timeout), will create one

```

**Comment by rajivml** [comment]
delete istio namespace and tried again, same error 

```
2020-07-28T20:33:07.185868Z     info    No certificates specified, skipping DNS certificate controller
2020-07-28T20:33:07.186492Z     info    CRD controller watching namespaces ""
2020-07-28T20:33:07.186663Z     info    Ingress controller watching namespaces ""
2020-07-28T20:33:07.186753Z     warn    Config Store &{0xc000ab2b40 cluster.local 0xc00028e580 0xc000806a50 0xc000150ea0 []} cannot track distribution in aggregate
2020-07-28T20:33:07.186766Z     info    Adding Kubernetes registry adapter
2020-07-28T20:33:07.186771Z     info    Service controller watching namespace "" for services, endpoints, nodes and pods, refresh 1m0s
2020-07-28T20:33:07.186834Z     info    JWT policy is third-party-jwt
2020-07-28T20:33:07.186878Z     info    Use self-signed certificate as the CA certificate
2020-07-28T20:33:37.187446Z     info    pkica   Failed to get secret (error: Get https://10.0.0.1:443/api/v1/namespaces/istio-system/secrets/istio-ca-secret: dial tcp 10.0.0.1:443: i/o timeout), will create one
2020-07-28T20:34:07.397189Z     error   pkica   Failed to write secret to CA (error: Post https://10.0.0.1:443/api/v1/namespaces/istio-system/secrets: dial tcp 10.0.0.1:443: i/o timeout). Abort.
2020-07-28T20:34:07.397233Z     error   failed to create discovery service: enableCA: failed to create a self-signed Citadel: failed to create CA due to secret write error
Error: failed to create discovery service: enableCA: failed to create a self-signed Citadel: failed to create CA due to secret write error
```
